packages/ripple-binary-codec/src/utils.ts

```diff
@@ -0,0 +1,149 @@
+// Don't know why this comes from NodeJS as it is valid in the browser
+import TypedArray = NodeJS.TypedArray
+
+//
+/**
+ * Writes value to array at the specified offset. The value must be a valid unsigned 8-bit integer.
+ * @param array Uint8Array to be written to
+ * @param value Number to be written to array.
+ * @param offset plus the number of bytes written.
+ */
+export function writeUInt8(
+  array: Uint8Array,
+  value: number,
+  offset: number,
+): void {
+  value = Number(value)
+  array[offset] = value
+}
+
+/**
+ * Writes value to array at the specified offset as big-endian. The value must be a valid unsigned 16-bit integer.
+ * @param array Uint8Array to be written to
+ * @param value Number to be written to array.
+ * @param offset plus the number of bytes written.
+ */
+export function writeUInt16BE(
+  array: Uint8Array,
+  value: number,
+  offset: number,
+): void {
+  value = Number(value)
+
+  array[offset++] = value >>> 8
+  array[offset++] = value
```

Comment:
This modifies the offset parameter - I'd prefer to see `(offset)` and `(offset + 1)` instead to avoid any confusion (although I know that in JavaScript it's pass-by-value instead of pass-by-reference, it took me a sec to verify whether this had side-effects or not)

I also think it's easy to misread this as `offset + 1` and `offset + 2` since the prepended ++offset is easily confused with appended offset++.

```suggestion
  array[offset] = value >>> 8
  array[offset + 1] = value
```

Reply:
(Unless I'm wrong and there is a side effect from modifying the parameter which ends up being used - in which case that's very surprising, and out of line with what the similar functions here do)